fs_src/script.js

```diff
@@ -846,12 +846,21 @@ function refreshUI() {
 }
 
 function setValueIfNotModified(e, newValue) {
-  if (e.lastSetValue !== undefined && e.lastSetValue !== e.value) return;
+  // do not update the value of the input field if
+  if (e.selected ||                    // the user has selected / highlighted the input field OR
+    e.lastSetValue === e.value ||      // the value has not been changed by the user OR
+    e.lastSetValue !== undefined       // a value has previously been set AND
```

Comment:
please put parens around && when combined with ||: `a || (b && c)` so reader does not need to remember operator precedence rules in mind when reading.

also, please leave operator at the end:

```
  a ||
  (b &&
   c)
```